src/messaging/hooks.ts

```
@@ -0,0 +1,13 @@
+import { WASocket } from 'baileys';
+import { updateMetaGroup } from '../models/group.ts';
+
+export async function socketHooks(sock: WASocket) {
+ setInterval(async () => {
+  try {
+   const data = await sock.groupFetchAllParticipating();
+   for (const [jid, metadata] of Object?.entries(data)) {
```

Biome (1.9.4)

[error] Unsafe usage of optional chaining.

If it short-circuits with 'undefined' the evaluation will throw TypeError here:


(lint/correctness/noUnsafeOptionalChaining)